pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -374,6 +384,50 @@ func (r *ReconcileStorageCluster) ensureCephObjectStores(instance *ocsv1.Storage
 	return nil
 }
 
+// newExternalCephObjectStoreInstances returns a set of CephObjectStores
+// needed for external cluster mode
+func (r *ReconcileStorageCluster) newExternalCephObjectStoreInstances(
+	initData *ocsv1.StorageCluster, rgwEndpoint string, reqLogger logr.Logger) ([]*cephv1.CephObjectStore, error) {
+	rgwEndpoint = strings.TrimSpace(rgwEndpoint)
+	var gateWay cephv1.GatewaySpec
+	if rgwEndpoint != "" {
```

Comment:
I don't think it can be empty, the UI will not accept that, or perhaps are you trying to catch a manual deployment issue?

Reply:
@leseb , there was discussion regarding making the `rgw-endpoint` optional and this function should serve both the needs.